lib/restclient.rb

```diff
@@ -90,8 +90,23 @@ def self.options(url, headers={}, &block)
     Request.execute(:method => :options, :url => url, :headers => headers, &block)
   end
 
-  class << self
-    attr_accessor :proxy
+  # A global proxy URL to use for all requests. This can be overridden on a
+  # per-request basis by passing `:proxy` to RestClient::Request.
+  def self.proxy
+    @proxy
+  end
+  def self.proxy=(value)
+    @proxy = value
+    @proxy_set = true
+  end
+
+  # Return whether RestClient.proxy was set explicitly. We use this to
+  # differentiate between no value being set and a value explicitly set to nil.
+  #
+  # @return [Boolean]
+  #
+  def self.proxy_set?
+    !!@proxy_set
```

Comment:
@darrenleeweber `!!` is a fairly common idiom for coercing any object to a Boolean, which is what we promise to return from this predicate. Some style guides discourage its use, but it is very concise if you recognize it.

http://codebytes.us/ternary-operator-and-the-double-bang-in-ruby/
http://stackoverflow.com/questions/524658/what-does-mean-in-ruby